app/resources/admin.html.template

```diff
@@ -138,35 +178,80 @@
     var title_col = row.insertCell(cur_col++);
     var title = document.createElement('input');
     title.type = 'text';
-    title.id = 'language_title_' + iteration;
+    title.id = 'repo_titles_language_value_' + iteration;
     title_col.appendChild(title);
-
     return iteration;
   }
 
+  function add_language_span(table_id, language) {
+    var table = $(table_id + '_table');
+    var iteration = table.rows.length;
+    var row = table.insertRow(iteration);
+    var cur_col = 0;
+  
+    var selected_language_col = row.insertCell(cur_col++);
+    var selected_language = document.createElement('span');
+    var selected_language_text = document.createTextNode(language);
+    selected_language.className = 'language_selected_' + iteration;
+    selected_language.appendChild(selected_language_text);
+    selected_language_col.style.verticalAlign = 'top';
+    selected_language_col.appendChild(selected_language);
+
+    var title_col = row.insertCell(cur_col++);
+    var title = document.createElement('textarea');
+    title.cols = 72;
+    title.id = table_id + '_language_value_' + iteration;
+    title_col.appendChild(title);
+    return iteration; 
+  }
+
+  function add_another_language() {
+    add_language();
+    for(var id_num = 1; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++){
+      add_language_span(CUSTOM_MESSAGE_TABLE_IDS[id_num], 'not selected');
+    }
+  }
+
   /* Hides a language from the user's view and marks it so that it will not be
    * submitted.
    */
   function remove_language(row_number) {
-    var table = $('language_table');
-    var row = table.rows[row_number];
-    row.hidden = true;
-    $('language_selection_' + row_number).value = 'unused';
+    for (var id_num = 0; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++) {
+      var table_name = CUSTOM_MESSAGE_TABLE_IDS[id_num];
+      var table = $(table_name + '_table');
+      var row = table.rows[row_number];
+      row.hidden = true;
+      if (table_name == 'repo_titles') {
+        $(table_name + '_language_selection_' + row_number).value = 'unused';
+      } else {
+        document.getElementsByClassName('language_selected_' + row_number).value
+            = 'unused';
+      }
+    }
   }
 
   // Adds rows to the languages table for each language already in the database
   function add_initial_languages() {
     // Saved languages and titles for this repository
+    var all_view_config_json = {{all_view_config_json|safe|default:'[]'}};
```

Comment:
Can you rename this variable to just all_view_config?  i.e.,
```
var all_view_config = {{all_view_config_json|safe|default:'[]'}};
```
Because the value of the variable is just a JavaScript object, no longer a JSON string.